lua/possession/config.lua

```diff
@@ -153,6 +153,11 @@ local function fix_compatibility(opts)
 
     local autoload = vim.tbl_get(opts, 'autoload')
     if type(autoload) == 'table' then
+        local utils = require('possession.utils')
+        utils.warn(
+            'Possession plugin: Usage of config setting `autoload = { cwd = %s }` is deprecated. Refer to docs for new setting values.',
```

Comment:
Do you know a nice way to split this string so it doesn't break Luacheck? 

https://stackoverflow.com/a/21205005 suggests `\z` which looks to work for me, but it also says that was introduced in Lua 5.2, and the Neovim docs say it uses Lua 5.1, so I'm not sure if this will work on older Neovim versions.

Alternatively, we could perhaps mark this line to be ignored by Luacheck somehow?

Reply:
You could probably just use string concatenation:
```lua
utils.warn('First part...' ..
    '...second parth')
```
Both less readable and less performant, but would probably work.

But because this is deprecation I would suggest using [vim.deprecate()](https://neovim.io/doc/user/lua.html#vim.deprecate()), e.g.
```lua
vim.deprecate('`setup.autoload.cwd = true`', '`autoload = "..."`', 'in the future', 'possession')
```
(we don't version this plugin so I'm not sure what to put as "version")

Reply:
Changed to use `vim.deprecate`.